addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/XxeScanRule.java

```diff
@@ -405,4 +453,29 @@ static String createLfrPayload(String requestBody) {
         sb.append(requestBody.substring(endIdx));
         return sb.toString();
     }
+
+    private int getTotalTagCount(String requestBody) {
+
+        int count = 0;
+        Matcher matcher = tagPattern.matcher(requestBody);
+        while (matcher.find()) {
+            count++;
+        }
+        return count;
+    }
+
+    static String createTagSpecificLfrPayload(String requestBody, int tagIdx) {
+        StringBuilder sb = new StringBuilder(ATTACK_HEADER);
+
+        Matcher matcher = tagPattern.matcher(requestBody);
```

Comment:
It would be better to keep the `Matcher` to not have to `find` every time from the start.